frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -1543,4 +1561,68 @@ private void setDefaultRequestAttributes(RenderRequest request) {
                 p -> nullToEmptyString(p.getState()));
         return isAscending ? comparator : comparator.reversed();
     }
+
+	private Object[] getCyclicProjectPresenceAndLastProjectInCycle(Project project, ProjectService.Iface client,
+            User user, Map<String, String> linkedProjectPath) throws TException {
+
+        Iterator<String> linkedProjectIterator = project.getLinkedProjects().keySet().iterator();
+
+        while (linkedProjectIterator.hasNext()) {
+            String linkedProjectId = linkedProjectIterator.next();
+            Project linkedProject = client.getProjectByIdForEdit(linkedProjectId, user);
+            String projectName = linkedProject.getName();
+            String projectVersion = linkedProject.getVersion();
+            String projFullName = getFullProjectName(projectName, projectVersion);
+            if (linkedProjectPath.containsKey(linkedProjectId)) {
+                return new Object[] { Boolean.TRUE, projFullName };
+            }
+
+            linkedProjectPath.put(linkedProjectId, projFullName);
+
+            Object[] cyclicLinkedProjectPresenceAndLastProjectInCycle = getCyclicProjectPresenceAndLastProjectInCycle(linkedProject,
+                   client, user, linkedProjectPath);
+            boolean isCyclicLinkedProjectPresent = (Boolean) cyclicLinkedProjectPresenceAndLastProjectInCycle[0];
+
+            if (isCyclicLinkedProjectPresent) {
+                return cyclicLinkedProjectPresenceAndLastProjectInCycle;
+            }
+            linkedProjectPath.remove(linkedProjectId);
+        }
+
+        return new Object[] { Boolean.FALSE, null };
+    }
+
+	private String getFullProjectName(String projectName, String projectVersion) {
```

Comment:
Logic is moved to ProjectDatabaseHandler , so that both UI and REST will be handled at common point.
Used printName(Project project) method present in SW360Utils class.
Code Formatting done.